### ISSUE TITLE
Tilt Bluetooth failing to open socket

### ISSUE THREAD

**Issue by TChilderhose** [thorrak/fermentrack#479]
**TL;DR I think you need to install `libbluetooth-dev` before compiling Python 3.7.7 for a fresh install.**

OS: Stretch
Fermentrack Branch: dev (Commit Date: | May 19, 2020, 10:07 a.m.)

Just did a fresh install of fermentrack last night/this morning. Had no errors so started configuring everything and noticed it's not detecting my TILT. 

Checked the logs and it's just a whole bunch of this stack trace. 

`/home/fermentrack/fermentrack/log/tilt--stderr.log`
```
Traceback (most recent call last):
  File "/home/fermentrack/fermentrack/gravity/tilt/tilt_monitor_aio.py", line 155, in <module>
    mysocket = aiobs.create_bt_socket(mydev)
  File "/home/fermentrack/venv/lib/python3.7/site-packages/aioblescan/aioblescan.py", line 1280, in create_bt_socket
    sock = socket.socket(family=socket.AF_BLUETOOTH,
AttributeError: module 'socket' has no attribute 'AF_BLUETOOTH'
Sentry is attempting to send 0 pending error messages
Waiting up to 2 seconds
Press Ctrl-C to quit
```

Looking at the Python source, it looks like it needs some bluetooth defines 
`Modules\sockemodule.c`
```
...
451: #if (defined(HAVE_BLUETOOTH_H) || defined(HAVE_BLUETOOTH_BLUETOOTH_H)) && !defined(__NetBSD__) && !defined(__DragonFly__)
452: #define USE_BLUETOOTH 1
...
7020: #ifdef USE_BLUETOOTH
7021:     PyModule_AddIntMacro(m, AF_BLUETOOTH);
...
```
Seems like `HAVE_BLUETOOTH_*` defines can be set multiple ways, but having `libbluetooth-dev` installed apparently can do it, but your script installs after python is compiled. I double checked that `libbluetooth-dev` is installed and ran ` fermentrack-tools/fix-environment.sh` forcing Python to re-compile and it's now fine. Tilt is connecting.

So yeah, will leave it to you to decide, but you just need to install `libbluetooth-dev` before compiling Python 3.7.7 for a fresh install.

Also, a few minor things
-  If I go to `/gravity/tilt/test/` it says `python3-dev` is `Not Installed`, which I believe is expected now that you're compiling it manually.
- How do I rename by Fermenter? I tried changing it in `/devices/1/manage/` but I can't get it to stick. I'm using an Arduino Uno if that effects anything.
- Thanks for your work on this project. Been manually maintaining my own BrewPi legacy fork and it's been showing its age haha.

**Comment by thorrak** [comment]
Hey there -- sorry for the delay in addressing this! You raised a couple of points, so I'll address them separately:

- I _think_ the installer now installs `libbluetooth-dev` prior to Python 3.7 in both `fix-environment.sh` and `install.sh` for both new and existing installs.

- Thanks for the heads up about `python3-dev` - I'll get that pulled out shortly

- I'll need to take a look at fermenter renaming. It should work just fine, but there's always the chance I missed something somewhere. I'll track this separately as #522
